plugins/experimental/ts_lua/ts_lua_transform.c

```diff
@@ -103,18 +110,29 @@ ts_lua_transform_handler(TSCont contp, ts_lua_http_transform_ctx *transform_ctx,
     transform_ctx->reserved.buffer = TSIOBufferCreate();
     transform_ctx->reserved.reader = TSIOBufferReaderAlloc(transform_ctx->reserved.buffer);
 
-    transform_ctx->upstream_bytes   = TSVIONBytesGet(input_vio);
+    if (empty_input == 0) {
```

Comment:
I'd be more tempted to use ternary operators in these cases, since the key is that something is always assigned. E.g.
```
transform_ctx->upstream_bytes = empty_input ? 0 : TSVIONBytesGets(input_vio);
```